src/coreclr/jit/importer.cpp

```diff
@@ -5599,6 +5601,14 @@ GenTree* Compiler::impCastClassOrIsInstToTree(
                 compCurBB->SetFlags(BBF_HAS_HISTOGRAM_PROFILE);
             }
         }
+        else if (!isCastClass && impIsCastHelperMayHaveProfileData(helper))
+        {
+            // Maybe the late-cast-expand phase will have a better luck expanding this cast.
+            // TODO: enable for cast-class as well.
+            call->gtCallMoreFlags |= GTF_CALL_M_CAST_CAN_BE_EXPANDED;
+            call->gtCastHelperILOffset = ilOffset;
+            setMethodHasExpandableCasts();
```

Comment:
Does it need to set it in the inline root?

Reply:
No, moved to morph (next to similar check for static cctors). It's not wrong, it just probably trigger the phase more often then